src/app/shared/ui/color-input/color-input.component.html

```diff
@@ -1,13 +1,17 @@
 <div>
   <span class="flex overflow-hidden rounded-md border border-neutral-200 shadow-sm dark:border-none dark:shadow-none">
     <label
+      [for]="readOnly() ? null : 'color'"
       [title]="tooltip()"
-      for="color"
     >
       <div
-        [class]="isColorLight() ? 'text-neutral-900' : 'text-neutral-100'"
+        [class]="
+          (isColorLight() ? 'text-neutral-900' : 'text-neutral-100') +
+          ' ' +
+          (readOnly() ? 'cursor-not-allowed' : 'cursor-pointer')
+        "
```

Comment:
Maybe something like that (to take some logic and concatenation out of the template)

```suggestion
       [class]="iconStyle()"
```
```

 protected iconStyle = computed(() => {
    const colorLight = this.isColorLight() ? 'text-neutral-900' : 'text-neutral-100';
    const readOnly = this.readOnly() ? 'cursor-not-allowed' : 'cursor-pointer';
    return [colorLight, readOnly].join(' ');
  })

```

Reply:
IMO it's always a trade off. I see why your suggestion is nicer to read (that's why I implemented it), but now you have to take a look at TS code to see the full style of the component.